lib/mix/tasks/docs.ex

```diff
@@ -184,6 +184,10 @@ defmodule Mix.Tasks.Docs do
       https://mydomain.org/user_or_team/repo_name/src/main/%{path}#cl-%{line}
       ```
 
+    * `:source_url_relative_path` - The absolute path pointing to where ExDoc should
+      consider the base of the repo. That is where the `%{path}` in the `:source_url_pattern`
+      starts. The default is the current working directory.
```

Comment:
Instead of adding a new option, what if we make `source_url_pattern` accept a function/MFA, that receives the path and the line, and then you can do whatever you want with it? Would that work for your use cases?

Reply:
hmm, yes that might work. I will give it a go and see if any issues pop up.

Reply:
Turns out that in #1529 someone else had the same idea as you and implemented it :) It just wasn't documented so I did not find that functionality. So I added documentation and removed this code.

Reply:
Beautiful!